unity-client/Assets/ABConverter/Client.cs

```diff
@@ -318,5 +333,151 @@ public static Core.State DumpAsset(string assetHash, string assetFilename, strin
+        /// <param name="abConverterCoreController">an instance of the ABCore</param>
+        /// <param name="items">an already-populated list of WearableItems</param>
+        /// <param name="OnWearableLoad">an action to be bind to the OnWearableLoad event on each wearable</param>
+        private static void DumpWearableQueue(ABConverter.Core abConverterCoreController, Queue<WearableItem> items, System.Action<UnityGLTF.GLTFSceneImporter> OnWearableLoad)
+        {
+            // We toggle the core's ABs generation off so that we execute that conversion here when there is no more items left.
+            abConverterCoreController.generateAssetBundles = false;
+            
+            if (items.Count == 0)
+            {
+                abConverterCoreController.ConvertDumpedAssets();
+
+                return;
+            }
+
+            Debug.Log("Building wearables... items left... " + items.Count);
+
+            var pairs = ExtractMappingPairs(new List<WearableItem>() { items.Dequeue() });
+            
+            UnityGLTF.GLTFImporter.OnGLTFWillLoad += OnWearableLoad;
+
+            abConverterCoreController.Convert(pairs.ToArray(),
+                (err) =>
+                {
+                    UnityGLTF.GLTFImporter.OnGLTFWillLoad -= OnWearableLoad;
+                    abConverterCoreController.CleanupWorkingFolders();
+                    DumpWearableQueue(abConverterCoreController, items, OnWearableLoad);
+                }, false);
+        }
+        
+        /// <summary>
+        /// Given a list of WearableItems, extracts and returns a list of MappingPairs
+        /// </summary>
+        /// <param name="wearableItems">A list of already-populated WearableItems</param>
+        /// <returns>A list of the extracted Wearables MappingPairs</returns>
+        private static List<ContentServerUtils.MappingPair> ExtractMappingPairs(List<WearableItem> wearableItems)
+        {
+            var result = new List<ContentServerUtils.MappingPair>();
+
+            foreach (var wearable in wearableItems)
+            {
+                foreach (var representation in wearable.representations)
+                {
+                    foreach (var datum in representation.contents)
+                    {
+                        result.Add(datum);
+                    }
+                }
+            }
+
+            return result;
+        }
+
+        /// <summary>
+        /// Given a base-url to fetch wearables collections, returns a list of all the WearableItems
+        /// </summary>
+        /// <param name="url">base-url to fetch the wearables collections</param>
+        /// <returns>A list of all the WearableItems found</returns>
+        private static List<WearableItem> GetAvatarMappingList(string url)
+        {
+            List<WearableItem> result = new List<WearableItem>();
+
+            UnityWebRequest w = UnityWebRequest.Get(url);
+            w.SendWebRequest();
+
+            while (!w.isDone) { }
```

Comment:
This can block the main thread if not used inside a coroutine, are we having this into consideration?

Reply:
Yes, this is on purpose, when we dump assets with the editor AB converter everything is done sequentially.